do/domains.go

```diff
@@ -157,20 +174,54 @@ func (ds *domainsService) DeleteRecord(domain string, id int) error {
 	return err
 }
 
-func (ds *domainsService) EditRecord(domain string, id int, drer *godo.DomainRecordEditRequest) (*DomainRecord, error) {
-	dr, _, err := ds.client.Domains.EditRecord(context.TODO(), domain, id, drer)
+// domainRecordRoot is the root of an individual Domain Record response.
+//
+// Copied from godo.
+type domainRecordRoot struct {
+	DomainRecord *DomainRecord `json:"domain_record"`
+}
+
+func (ds *domainsService) EditRecord(domain string, id int, drer *DomainRecordEditRequest) (*DomainRecord, error) {
+	if len(domain) < 1 {
+		return nil, godo.NewArgError("domain", "cannot be an empty string")
+	}
+	if id < 1 {
+		return nil, godo.NewArgError("id", "cannot be less than 1")
+	}
+	if drer == nil {
+		return nil, godo.NewArgError("editRequest", "cannot be nil")
+	}
+
+	path := fmt.Sprintf("v2/domains/%s/records/%d", domain, id)
+	req, err := ds.client.NewRequest(context.TODO(), http.MethodPut, path, drer)
 	if err != nil {
 		return nil, err
 	}
 
-	return &DomainRecord{DomainRecord: dr}, nil
+	root := new(domainRecordRoot)
+	if _, err := ds.client.Do(context.TODO(), req, root); err != nil {
+		return nil, err
+	}
+	return root.DomainRecord, nil
 }
 
-func (ds *domainsService) CreateRecord(domain string, drer *godo.DomainRecordEditRequest) (*DomainRecord, error) {
-	dr, _, err := ds.client.Domains.CreateRecord(context.TODO(), domain, drer)
+func (ds *domainsService) CreateRecord(domain string, drer *DomainRecordEditRequest) (*DomainRecord, error) {
+	if len(domain) < 1 {
+		return nil, godo.NewArgError("domain", "cannot be empty string")
+	}
+	if drer == nil {
+		return nil, godo.NewArgError("createRequest", "cannot be nil")
+	}
+
+	path := fmt.Sprintf("v2/domains/%s/records", domain)
```

Comment:
nit: can we pull this as a const?

Reply:
What here would be a `const`?

Reply:
const domainsRecords = `v2/domains/%s/records`

Reply:
Gotcha, will do